bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyBasicDiscoveryService.java

```diff
@@ -0,0 +1,197 @@
+        logger.debug("Adding discovered thing with id {}", uid.toString());
+        properties.put(PROPERTY_MAC_ADDRESS, address);
+        String thingLabel = "Shelly BLU " + model + " (" + serviceName + ")";
+        DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                .withRepresentationProperty(PROPERTY_DEV_NAME).withLabel(thingLabel).build();
+        thingDiscovered(result);
+    }
+
+    public void discoveredResult(DiscoveryResult result) {
+        thingDiscovered(result);
+    }
+
+    public void unregisterDeviceDiscoveryService() {
+        if (discoveryService != null) {
+            discoveryService.unregister();
+        }
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+        unregisterDeviceDiscoveryService();
+    }
+
+    public static @Nullable DiscoveryResult createResult(boolean gen2, String hostname, String ipAddress,
+            ShellyBindingConfiguration bindingConfig, HttpClient httpClient, ShellyTranslationProvider messages) {
+        ThingUID thingUID = null;
+        ShellyDeviceProfile profile;
+        ShellySettingsDevice devInfo;
+        ShellyApiInterface api = null;
+        boolean auth = false;
+        String mac = "";
+        String model = "";
+        String mode = "";
+        String name = hostname;
+        String deviceName = "";
+        String thingType = "";
+        Map<String, Object> properties = new TreeMap<>();
+
+        try {
+
+            ShellyThingConfiguration config = fillConfig(bindingConfig, ipAddress);
+            api = gen2 ? new Shelly2ApiRpc(name, config, httpClient) : new Shelly1HttpApi(name, config, httpClient);
+            api.initialize();
+            devInfo = api.getDeviceInfo();
+            mac = getString(devInfo.mac);
+            model = devInfo.type;
+            auth = getBool(devInfo.auth);
+            if (name.isEmpty()) {
+                name = devInfo.hostname;
+            }
+            if (devInfo.name != null) {
+                deviceName = devInfo.name;
+            }
+
+            thingType = substringBeforeLast(name, "-");
+            profile = api.getDeviceProfile(thingType, devInfo);
+            api.close();
+            deviceName = profile.name;
+            mode = devInfo.mode;
+            properties = ShellyBaseHandler.fillDeviceProperties(profile);
+
+            // get thing type from device name
+            thingUID = ShellyThingCreator.getThingUID(name, model, mode, false);
+        } catch (ShellyApiException e) {
+            ShellyApiResult result = e.getApiResult();
+            if (result.isHttpAccessUnauthorized()) {
+                // create shellyunknown thing - will be changed during thing initialization with valid credentials
+                thingUID = ShellyThingCreator.getThingUID(name, model, mode, true);
+            }
+        } catch (IllegalArgumentException | IOException e) { // maybe some format description was buggy
+
```

Comment:
Very minor, but instead of ignoring, you could log to debug or trace

Reply:
This is a static function so logger.debug is not available

Reply:
What about
`LoggerFactory.getLogger(ShellyBasicDiscoveryService.class).debug(……..`

Reply:
oh, cool; was not aware of this
changed